core/src/main/java/org/geysermc/geyser/session/GeyserSession.java

```diff
@@ -936,19 +936,7 @@ private void connectDownstream() {
 
         // This isn't a great solution, but... we want to make sure the finish configuration packet cannot be sent
         // before the KnownPacks packet.
-        this.downstream.getSession().addListener(new ClientListener(ProtocolState.LOGIN, loginEvent.transferring()) {
-            @Override
-            public void packetReceived(Session session, Packet packet) {
-                if (protocol.getState() == ProtocolState.CONFIGURATION) {
-                    if (packet instanceof ClientboundFinishConfigurationPacket) {
-                        // Prevent
-                        GeyserSession.this.ensureInEventLoop(() -> GeyserSession.this.sendDownstreamPacket(new ServerboundFinishConfigurationPacket()));
-                        return;
-                    }
-                }
-                super.packetReceived(session, packet);
-            }
-        });
+        this.downstream.getSession().addListener(new ClientListener(ProtocolState.LOGIN, loginEvent.transferring()));
```

Comment:
This line can be completely removed, and the above comment can also be removed, and we can re-enable useDefaultListeners.